packages/binding-coap/src/coap-server.ts

```diff
@@ -337,387 +335,478 @@ export default class CoapServer implements ProtocolServer {
+
+        return uri;
+    }
+
+    private handleThingsRequest(method: string, res: OutgoingMessage) {
+        if (method !== "GET") {
+            this.sendMethodNotAllowedResponse(res);
+            return;
+        }
+
+        const payload = JSON.stringify(this.getThingDescriptionPayload());
+        this.sendContentResponse(res, payload, ContentSerdes.DEFAULT);
+    }
+
+    private async handlePropertyRequest(
+        thing: ExposedThing,
+        affordanceKey: string,
+        req: IncomingMessage,
+        res: OutgoingMessage,
+        contentType?: string
+    ) {
+        const property = thing.properties[affordanceKey];
+
+        if (property == null) {
+            this.sendNotFoundResponse(res);
+            return;
+        }
+
+        switch (req.method) {
+            case "GET":
+                if (req.headers.Observe == null) {
+                    this.handleReadProperty(property, req, contentType, thing, res, affordanceKey);
+                } else {
+                    this.handleObserveProperty(req, thing, res, affordanceKey);
+                }
+                break;
+            case "PUT":
+                if (property.readOnly === true) {
+                    this.sendErrorResponse(res, "4.00", "Property readOnly");
+                    return;
+                }
+
+                this.handleWriteProperty(property, req, contentType, thing, res, affordanceKey);
+                break;
+            default:
+                this.sendMethodNotAllowedResponse(res);
+        }
+    }
+
+    private async handleReadProperty(
+        property: PropertyElement,
+        req: IncomingMessage,
+        contentType: string,
+        thing: ExposedThing,
+        res: OutgoingMessage,
+        affordanceKey: string
+    ) {
+        try {
+            const interactionOptions = this.createInteractionOptions(
+                property.forms,
+                property.uriVariables,
+                thing,
+                req,
+                contentType
             );
-        });
+            const content = await thing.handleReadProperty(affordanceKey, interactionOptions);
+            this.streamContentResponse(res, content);
+        } catch (err) {
+            error(`CoapServer on port ${this.getPort()} got internal error on read '${req.url}': ${err.message}`);
+            this.sendErrorResponse(res, "5.00", err.message);
+        }
+    }
 
-        const requestUri = req.url;
-        let contentType = req.headers["Content-Format"] as string;
+    private async handleObserveProperty(
+        req: IncomingMessage,
+        thing: ExposedThing,
+        res: OutgoingMessage,
+        affordanceKey: string
+    ) {
+        const listener = async (content: Content) => {
+            try {
+                this.streamContentResponse(res, content, { end: true });
+            } catch (err) {
+                error(`CoapServer on port ${this.getPort()} got internal error on read '${req.url}': ${err.message}`);
```

Comment:
```suggestion
                error(`CoapServer on port ${this.getPort()} got internal error on observe '${req.url}': ${err.message}`);
```

Shouldn't it be observe?